### ISSUE TITLE
[P1][ARU][P04P/P04Q-MODEL-ID] Torch failures silently relabel MLP/GBT predictions as CNN/ResNet methods

### ISSUE THREAD

**Issue by SzeChunYiu** [SzeChunYiu/ccb-testbeam#1126]
## Atom definition / severity

**Severity:** P1 for model-comparison, multiplicity, portability, and provenance; P2 for the already committed P04p run if every Torch fold is independently proven to have executed successfully. Elevate to P0 if a fallback-labelled method can become an authorising winner.

**Observed current main:** `741725c85df04c20238c0500f5e1b944fe3f9161`.

**Parents:** P04p/P04q model-selection contract (`AUD-P04P-001`), environment/provenance identity, run-heldout split. **Children:** method ranking, cross-stave transfer, reported model-family multiplicity, claim-ledger method identity.

## Exact software/data contract

P04p and P04q advertise distinct methods:

- `mlp`
- `gradient_boosted_trees`
- `cnn_1d`
- `wavegate_resnet`

The Torch methods are attempted in one `try` block. On **any exception**, current code executes:

```python
fold["prob_cnn_1d"] = fold["prob_mlp"]
fold["prob_wavegate_resnet"] = fold["prob_gradient_boosted_trees"]
```

and then computes ordinary flags/metrics under the names `cnn_1d` and `wavegate_resnet`.

Thus the output contract can assert four model identities while only two distinct predictive functions were evaluated.

The committed P04p `result.json` records only coarse `torch_available: true`; that field proves that Torch was importable in the environment, not that **each fold and each named Torch model completed successfully**. The fallback exception is printed to stdout but is not a structured fold-level artifact.

## Model-identity invariant

Let `M` be the declared method and `P_M(x)` its prediction vector. A valid benchmark record must satisfy either

`execution_state(M)=SUCCESS` and `P_M` is produced by the declared implementation,

or

`execution_state(M)=FAILED/UNAVAILABLE` and no scientific metric/rank is attributed to `M`.

It is invalid to perform the map

`FAILED(CNN) -> P_MLP` while retaining label `CNN`.

An explicit fallback can be scientifically valid only under a **different model ID**, e.g. `cnn_unavailable_fallback_mlp`, and that fallback must normally be excluded from model-family comparison because it adds no independent method.

## Competing execution semantics

- **H1 fail closed:** failed model receives `FAILED_MODEL_EXECUTION`; no metrics/rank.
- **H2 explicit operational fallback:** use MLP/GBT for deployment continuity but label the effective model exactly and do not count it as CNN/ResNet evidence.
- **H3 retry/recover:** rerun the same declared model under a controlled resource policy and record both failure and retry provenance.
- **H4 current silent alias:** prediction bytes come from another model but are published under the failed model name. This is eliminated by provenance/model-identity semantics.

## Deterministic falsifier

Monkeypatch `fit_torch_classifier()` to raise a deliberate exception on one held-out fold. Current expected behavior is then exactly

```text
prob_cnn_1d == prob_mlp
prob_wavegate_resnet == prob_gradient_boosted_trees
```

while all four method names remain in the benchmark table. This is a zero-data known-answer test of the defect.

A stronger invariant test should hash each prediction vector and assert that identical vectors across supposedly different methods are either scientifically expected and explained or marked as an alias/fallback rather than independent model evidence.

### 1. Domain/modeling lead
Define which methods are scientifically distinct hypotheses and which are operational fallbacks. A waveform CNN and a tabular MLP are different mechanism classes only if their declared code actually generated the predictions.

### 2. Adversarial mechanism/software reviewer
Inject import errors, CUDA/CPU errors, OOM, NaN training, one-model-only failure, and mid-fold exceptions. Verify that no exception path can create a successful scientific method row under the wrong identity.

### 3. Independent statistics/validation reviewer
Model-family multiplicity and winner selection must use the set of **successfully executed distinct models**, not aliases. Recompute selection if any historical fold failed. If a method fails in only some runs, do not pool heterogeneous effective models under one name; either fail the full method or define a predeclared missing-fold policy.

### 4. Claims/provenance reviewer
Persist per fold: requested model, effective model, implementation/source hash, library versions, seed, execution state, exception type/message digest, prediction hash, and retry/fallback state. Public tables must visibly mark unavailable methods.

## Discriminating tests / negative controls

1. Force both Torch methods to throw: benchmark must contain `FAILED`, not copied MLP/GBT metrics.
2. Force only CNN to fail: WaveGate result remains independent; CNN does not inherit MLP silently.
3. Torch importable but training raises: `torch_available=true` must not be interpreted as successful execution.
4. Successful deterministic tiny-network fixture: requested/effective model IDs match and prediction hash is stable under fixed seed/environment.
5. Alias sentinel: intentionally return identical predictions from two methods and require explicit `prediction_alias_detected` provenance.
6. Re-run historical P04p/P04q and inspect captured stdout/logs where available for any Torch failure messages; if logs are unavailable, regenerate under content-addressed provenance.

## Implementation plan

1. Replace broad fallback assignment with structured `ModelExecutionResult` carrying `requested_model`, `effective_model`, `status`, `probabilities`, `error`.
2. Default scientific benchmarking to fail closed for failed methods.
3. If an operational fallback is desired, emit it under a new explicit method ID and exclude it from independent model ranking/multiplicity unless the selection contract says otherwise.
4. Move CNN and WaveGate into separate try/exception scopes so one failure cannot erase the other.
5. Write per-fold execution records to JSON/CSV and include prediction SHA-256.
6. Add a ranking gate: only `SUCCESS` + model-ID-matched rows are eligible.
7. Regenerate P04p/P04q model tables if any historical execution state cannot be proven.

## Acceptance / rejection criteria

- [ ] A model failure cannot produce metrics under that model's name from another estimator's predictions.
- [ ] Torch importability and successful per-fold execution are distinct fields.
- [ ] One Torch model can fail without silently replacing the other.
- [ ] All ranked methods have requested/effective model identity and prediction hashes.
- [ ] Alias/fallback methods do not inflate model-family multiplicity.
- [ ] Historical results are either proven all-fold successful or regenerated/quarantined.

Do **not** close by noting that the committed P04p artifact says `torch_available=true`; import availability is not fold-level execution evidence.

## Required data/MC, uncertainty, weights

The primary defect is testable without detector data using injected exceptions. Historical-impact closure requires only the existing P04p/P04q raw data and seeds/configs. No MC/generator weights are involved. Statistical uncertainty is secondary; first establish the exact method set before any bootstrap/model-selection calculation.

## Claim/wiki consequences

CL-015 remains `GATED` independently. P04p/P04q reports must not count a failed/aliased Torch method as an evaluated independent model family. If an aliased method ever affected rank, multiplicity, or winner selection, regenerate the relevant claim surfaces.

## Exact handoff

Write a unit test that monkeypatches `fit_torch_classifier` to raise on a fold and asserts the current code produces copied predictions under wrong names. Then replace the fallback with typed execution states before changing any model hyperparameters.

**Comment by SzeChunYiu** [comment]
Closing under campaign ledger evidence (#1218 compliant manual close).

- Status: FIXED (fail-closed contract / ADR landed)
- Merged PR(s): #1242
- Ledger: `reports/mc_validation/execution/ISSUE_CAMPAIGN_LEDGER.md`

If this was intended as physics-BLOCKED rather than code-fixed, reopen and mark BLOCKED with ADR.
